RHDEVDOCS-3658- RN Reformat (#41059)
openshift/openshift-docs: modules/cluster-logging-release-notes-5.3.z.adoc

//Z-stream Release Notes by Version

[id="cluster-logging-release-notes-5-3-3"]
= OpenShift Logging 5.3.3
This release includes link:https://access.redhat.com/errata/RHSA-2022:0227[RHSA-2022:0227 OpenShift Logging Bug Fix Release 5.3.3]

[id="openshift-logging-5-3-3-bug-fixes"]
== Bug fixes

> * Before this update, changes to the metrics dashboards had not yet been deployed because the cluster-logging-operator did not correctly compare existing and desired configmaps containing the dashboard.  This update fixes the logic by adding a the dashboard unique hash value to the object labels.(link:https://issues.redhat.com/browse/LOG-2066[LOG-2066])

Reviewer: ```suggestion
* Before this update, changes to the metrics dashboards had not yet been deployed because the cluster-logging-operator did not correctly compare existing and desired configmaps containing the dashboard. This update fixes the logic by adding a the dashboard unique hash value to the object labels.(link:https://issues.redhat.com/browse/LOG-2066[LOG-2066])
```

* This update changes the log4j dependency to 2.17.1 to resolve link:https://access.redhat.com/security/cve/CVE-2021-44832[CVE-2021-44832].(link:https://issues.redhat.com/browse/LOG-2102[LOG-2102])

== CVEs
[id="openshift-logging-5-3-3-CVEs"]
.Click to expand CVEs
[%collapsible]
====
* link:https://access.redhat.com/security/cve/CVE-2021-27292[CVE-2021-27292]
** link:https://bugzilla.redhat.com/show_bug.cgi?id=1940613[BZ-1940613]
* link:https://access.redhat.com/security/cve/CVE-2021-44832[CVE-2021-44832]
** link:https://bugzilla.redhat.com/show_bug.cgi?id=2035951[BZ-2035951]
====

[id="cluster-logging-release-notes-5-3-2"]
= OpenShift Logging 5.3.2
This release includes link:https://access.redhat.com/errata/RHSA-2022:0044[RHSA-2022:0044 OpenShift Logging Bug Fix Release 5.3.2]

[id="openshift-logging-5-3-2-bug-fixes"]
== Bug fixes

> * Before this update, Elasticsearch rejected logs from the eventrouter due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, existing indices will cause an warning/error within Kibana. The field ("kubernetes.event.metadata.resourceVersion") will cause an error until the removal or reindexing of existing indices. If this field is not in use in Kibana, then you can ignore the error message. For those with a delete policy, that policy will eventually remove old indices. When that occurs, the message will no longer generate. Those without a delete policy in place will need to manually reindex in order to remove the message.(link:https://issues.redhat.com/browse/LOG-2088[LOG-2088])

Reviewer: Apply any updates to LOG-2087 here as well.

> * Before this update, the OpenShift Logging Dashboard displayed the wrong pod namespace in the table which displays top producing and collected containers over the last 24 hours. With this update, the OpenShift Logging Dashboard displays the correct pod namespace.(link:https://issues.redhat.com/browse/LOG-2051[LOG-2051])

Reviewer: Does the following guidance from the IBM SG apply here:
"Use that, without a comma, to introduce a restrictive clause. A restrictive clause is essential to the
meaning of the sentence. If the restrictive clause is removed, the meaning of the sentence changes."
If so, consider:
```suggestion
* Before this update, the OpenShift Logging Dashboard displayed the wrong pod namespace in the table that displays top producing and collected containers over the last 24 hours. With this update, the OpenShift Logging Dashboard displays the correct pod namespace.(link:https://issues.redhat.com/browse/LOG-2051[LOG-2051])
```

> * Before this update, any situation with Elasticsearch OutputDefault defined for cluster logforwarding and Elasticsearch outputs did not have structured keys, the configuration was replacing the output secret with the default secret used to communicate to the default log store. With this update, the defined output secret is correctly used.(link:https://issues.redhat.com/browse/LOG-2046[LOG-2046])

Reviewer: ```suggestion
* Before this update, if the `outputDefaults` elements for Elasticsearch in the `ClusterLogForwarder` CR YAML file did not have structured keys, the configuration replaced the output secret with the default secret that was used to communicate with the default log store. With this update, the defined output secret is correctly used.(link:https://issues.redhat.com/browse/LOG-2046[LOG-2046])
```

[id="openshift-logging-5-3-2-CVEs"]
== CVEs
.Click to expand CVEs
[%collapsible]
====
* https://access.redhat.com/security/cve/CVE-2020-36327[CVE-2020-36327]
** https://bugzilla.redhat.com/show_bug.cgi?id=1958999[BZ-1958999]
* https://access.redhat.com/security/cve/CVE-2021-45105[CVE-2021-45105]
** https://bugzilla.redhat.com/show_bug.cgi?id=2034067[BZ-2034067]
* https://access.redhat.com/security/cve/CVE-2021-3712[CVE-2021-3712]
* https://access.redhat.com/security/cve/CVE-2021-20321[CVE-2021-20321]
* https://access.redhat.com/security/cve/CVE-2021-42574[CVE-2021-42574]
====

[id="cluster-logging-release-notes-5-3-1"]
= OpenShift Logging 5.3.1
This release includes link:https://access.redhat.com/errata/RHSA-2021:5129[RHSA-2021:5129 OpenShift Logging Bug Fix Release 5.3.1]

[id="openshift-logging-5-3-1-bug-fixes"]

> == Bug fixes

Reviewer: ```suggestion
== Bug fixes

```
Leave a space before bullet lists.

> * Before this update, the image included builder tools that are unnecessary at runtime. This update removes tools unnecessary at runtime. (link:https://issues.redhat.com/browse/LOG-1998[LOG-1998])

Reviewer: ```suggestion
* Before this update, the Fluentd container image included builder tools that were unnecessary at run time. This update removes those tools from the image. (link:https://issues.redhat.com/browse/LOG-1998[LOG-1998])
```
IBM SG:
```
run time, runtime noun
Write as two words to refer to the time at which an application runs (for example, "At run time, you can ...").
Write as one word to refer to the environment in which an application runs (for example, "The Java runtime is
controlled by ...").
```

* Before this update, the Logging dashboard displayed an empty CPU graph because of a reference to an invalid metric. With this update, the Logging dashboard displays CPU graphs correctly. (link:https://issues.redhat.com/browse/LOG-1925[LOG-1925])

> * Before this update, the Elasticsearch Prometheus exporter plugin was compiling index-level metrics using a high-cost query which had a negative performance impact on the processing Elasticsearch node. With this update, this query has been replaced with a low-impact query that will not impact performance.(link:https://issues.redhat.com/browse/LOG-1897[LOG-1897])

Reviewer: Duplicate of LOG-1970?
Check for other duplicates?
